test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -190,4 +193,117 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
+				return newCQ.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{kueue.ResourceInUseFinalizerName}))
+
+			ginkgo.By("New created workloads should not be admitted")
+			gomega.Eventually(func() *kueue.Admission {
+				newWL := testing.MakeWorkload("new-workload", ns.Name).Queue(queue.Name).Obj()
+				gomega.Expect(k8sClient.Create(ctx, newWL)).To(gomega.Succeed())
+				return newWL.Spec.Admission
+			}, framework.Timeout, framework.Interval).Should(gomega.BeNil())
+
+			ginkgo.By("Finish workloads")
+			gomega.Eventually(func() error {
+				var newWL kueue.Workload
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &newWL)).To(gomega.Succeed())
+				newWL.Status.Conditions = append(newWL.Status.Conditions, kueue.WorkloadCondition{
+					Type:   kueue.WorkloadFinished,
+					Status: corev1.ConditionTrue,
+				})
+				return k8sClient.Status().Update(ctx, &newWL)
+			}, framework.Timeout, framework.Interval).Should(gomega.BeNil())
+
+			ginkgo.By("The clusterQueue will be deleted")
+			gomega.Eventually(func() bool {
+				var newCQ kueue.ClusterQueue
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)
+				return err != nil && apierrors.IsNotFound(err)
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("Should not admit workloads in termination when updating resourceFlavors", func() {
+			ginkgo.By("Admit workloads")
+			admission := testing.MakeAdmission(cq.Name).Obj()
+			wl := testing.MakeWorkload("workload", ns.Name).Queue(queue.Name).Admit(admission).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+			defer func() {
+				gomega.Expect(framework.DeleteWorkload(ctx, k8sClient, wl)).To(gomega.Succeed())
+			}()
+
+			ginkgo.By("Delete clusterQueue")
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+			var newCQ kueue.ClusterQueue
+			gomega.Eventually(func() []string {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)).To(gomega.Succeed())
+				return newCQ.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{kueue.ResourceInUseFinalizerName}))
+
+			ginkgo.By("Create resourceFlavor")
+			resourceFlavor := testing.MakeResourceFlavor("resource-flavor").Obj()
+			gomega.Expect(k8sClient.Create(ctx, resourceFlavor)).To(gomega.Succeed())
+			defer func() {
+				gomega.Expect(framework.DeleteResourceFlavor(ctx, k8sClient, resourceFlavor)).To(gomega.Succeed())
+			}()
+
+			ginkgo.By("New created workloads should not be admitted")
+			gomega.Eventually(func() *kueue.Admission {
+				newWL := testing.MakeWorkload("new-workload", ns.Name).Queue(queue.Name).Obj()
+				gomega.Expect(k8sClient.Create(ctx, newWL)).To(gomega.Succeed())
+				return newWL.Spec.Admission
+			}, framework.Timeout, framework.Interval).Should(gomega.BeNil())
```

Comment:
similar to the comment above, the scheduler is not running so no one will actually attempt to admit the workload.